include/ssm-internal.h

```diff
@@ -232,4 +252,8 @@ void ssm_mem_init(void *(*alloc_page_handler)(void),
 
 /** @} */
 
+#ifndef NVALGRIND
+#include <valgrind/memcheck.h>
```

Comment:
We include valgrind-related macros if `NVALGRIND` is not defined (this is the compilation interface suggested by Valgrind documentation, inspired by assert.h and `NDEBUG`).